File: src/Goods.Domain/Settlements/Settlements.cs
namespace Goods.Domain.Products;

public class Settlements
{

    /*
        id uuid primary key not null,
	    settlementtype uuid NOT NULL,
	    name varchar NOT NULL,
	    population int NOT NULL,
	    region uuid NOT NULL,
	    foundationyear varchar(4) NOT NULL,
	    ishero bool NOT NULL,
	    averagehotelcost int NOT NULL,
     */
    
    
    public Guid Id { get; set; }
    public SettlementsTypes Type { get; set; }
    public String Name { get; set; }
    public Int32 Population {  get; set; }
    public Regions Region { get; set; }
    public Int32 FoundationYear { get; set; }
    public Boolean IsHero { get; set; }
    public Int32 AverageHotelCost { get; set; }

    public Settlements
    (
        Guid id,
        SettlementsTypes type,
        String name,
        Int32 population,
        Regions region,
        Int32 foundationYear,
        Boolean isHero,
        Int32 averageHotelCost
    )
    {
        Id = id;
        Type = type;
        Name = name;
        Population = population;
        Region = region;
        FoundationYear = foundationYear;
        IsHero = isHero;
        AverageHotelCost = averageHotelCost;
    }
}

File: src/Goods.BackOffice/client-app/src/domain/settlements/settlements.ts
import { Page } from '../../tools/types/page';
import { Regions } from '../regions/regions';
import { SettlementsTypes } from './settlementsTypes';
import { mapToRegion } from '../regions/regions';

export class Settlements {
	constructor(
		public readonly id: string,
		public readonly type: SettlementsTypes,
		public readonly name: string,
		public readonly population: number,
		public readonly region: Regions,
		public readonly foundationYear: number,
		public readonly isHero: boolean,
		public readonly averageHotelCost: number
	) { }
}

export function mapToSettlementsPage(data: any): Page<Settlements> {
	return Page.convert(data, mapToSettlement);
}

export function mapToSettlements(data: any[]): Settlements[] {
	return data.map(mapToSettlement);
}

export function mapToSettlement(data: any): Settlements {
	return new Settlements(data.id, data.type, data.name, data.population, mapToRegion(data.region), data.foundationYear, data.isHero, data.averageHotelCost);
}
